### PR TITLE
Security

### PR DESCRIPTION
**Pull request by morganelle**
- implementation of authorization, authentication and csrf protection
- new and refactored tests for new authz, authn and csrf protection
- updated README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ Collaborators: James Feore
 ## Deployment:
 Deployed on Heroku at https://rocky-inlet-61764.herokuapp.com/
 
+## Test instructions:
+Test should run on their own as dummy environmental variables are included in tests.py.
+
 
 ## Tests:
 ### Test plan
```
Comment (clair3st) [inline README.md]: Consider putting your tox output in a code block (ie triple backtick), might make it easier on the eye.

File: README.md [hunk 2]
```diff
@@ -20,33 +23,41 @@ Name                                      Stmts   Miss  Cover   Missing
...
+
+46 passed in 10.39 seconds
+
 
-27 passed in 2.12 seconds
-__________________________________________________________ 
-__________________________________________________________
   py27: commands succeeded
   py36: commands succeeded
+  congratulations :)
```
Comment (clair3st) [inline README.md]: You should consider adding a more step by step process for how to get a user up and running with this. Including commands and any environment variables that have to be set.

File: learning_journal/setup.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
     'ipython',
     'pyramid_ipython',
     'psycopg2',
+    'passlib',
 ]
```
Comment (clair3st) [inline learning_journal/setup.py:25]: Don't forget to add this to your tox requires

**Review by clair3st — COMMENTED** [review]
Overall, really well done with this. I just had a little trouble running tox because of the new dependancies missing from `tox.ini`

File: learning_journal/setup.py (hunk 3 above)
Comment (morganelle) [reply learning_journal/setup.py:25]: Updated!

File: README.md (hunk 1 above)
Comment (morganelle) [reply README.md]: Updated!

File: learning_journal/setup.py (hunk 3 above)
Comment (morganelle) [reply learning_journal/setup.py:25]: Updated!

File: README.md (hunk 2 above)
Comment (morganelle) [reply README.md]: Updated!

**Pull request merged**